### DISCUSSION TITLE
Suggestion: Static analysis / Adopting explicit rules for code style

### DISCUSSION THREAD

**Discussion by mthh** [neocarto/bertin#95] [Ideas]
This is just a suggestion but by Bertin v2 you could add explicit rules about code style (number of spaces for indent, location of brackets after an if/for, location of comments, etc.) and add static code analysis with something like [ESLint](https://en.wikipedia.org/wiki/ESLint).

ESLint helps identify problematic or confusing code patterns (see for example [here](https://eslint.org/docs/latest/rules/)) and helps to enforce a chosen code style. It integrates with most code editors (and is used by many projects) so problems can be seen when writing code.

This would help to guarantee the homogeneity of the code base (style of the code, development pattern used, etc.) especially for contributors who submit code through pull requests.

**Comment by neocarto** [comment]
Sounds good.
